### ISSUE TITLE
Exceptions 0.10

### ISSUE THREAD

**Issue by jacobstanley** [tmcgilchrist/entwine#15]
I can't build mafia with newer GHCs otherwise

**Comment by jacobstanley** [comment]
Just keeping my own fork that doesn't use entwine for the moment

**Comment by tmcgilchrist-da** [comment]
A new hackage release of entwine (https://hackage.haskell.org/package/entwine) should fix it

Which GHC versions were you trying to use?

**Comment by jacobstanley** [comment]
Cool, just wanted to build with ghc-8.6, seems that has to use a newer version of exceptions unless I was doing something wrong

**Comment by tmcgilchrist** [comment]
The other issue with 8.6 is supporting newer Cabal versions. 
I've been running with 2.2 locally for 6 months without any issues, so I think the Chaos test is finding something that doesn't occur very often in practice.

https://github.com/haskell-mafia/mafia/pull/234

**Comment by jacobstanley** [comment]
Yeah and I found that with 2.4 it spews warnings, so I'm running a version which changes all the commands to `v1-xxx` but it only works with `cabal-install >= 2.4`
